kb(loadercontainer):center in viewport (#943)
telerik/blazor-docs: knowledge-base/loader-container-center-in-viewport.md

---

> title: Center LoaderContainer in viewport

Reviewer: ```suggestion
title: Center LoaderContainer in Viewport
```

description: How to center the LoaderContainer in viewport
type: how-to

> page_title: Center LoaderContainer in viewport

Reviewer: page_title can be just a little longer to add 1-2 more keywords. If you think "page" will be more relevant, consider that instead - the customer used it as well.

```suggestion
page_title: Center LoaderContainer in the Browser Viewport
```

slug: loader-container-kb-center-in-viewport
position: 
tags: loadercontainer, center, viewport, long, page
ticketid: 1564614
res_type: kb
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>LoaderContainer for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

> We use the TelerikLoaderContainer component on our pages to let the user know that we're doing something in the background. Some of our pages are twice or three times the height of the user's viewport in their browser. When the user takes an action that could take some length of time we display the TelerikLoaderContainer but it always displays in the center of the page as a whole, not in the center of the browser's viewport. So, if the user is at the top or bottom of the page they won't see the loader animation.

Reviewer: Generally, there is no need to copy-paste everything the customer said. You can extract all the important keywords and use them in a shorter paragraph.

How to display the LoaderContainer in the center of the viewport?

> <br/>

Reviewer: ```suggestion
```

> How to center the LoaderContainer in the viewport regardless of how long the page actually is?

Reviewer: This sentence uses the same main keywords as the previous one, just in different order. When you say the same thing multiple times, try to use different keywords.

## Solution

> You can achieve that by using some custom CSS to modify the LoaderContainer position to "fixed". This will center the component in respect to the whole page width.

Reviewer: ```suggestion
Use custom CSS to override the LoaderContainer `position` style to `fixed`. This will center the component in respect to the whole page width.
```

In addition, you may also adjust the position of the inner container - the one that holds the text and the animation. Thus, you can center it in respect to the "main" element and not the whole page.

The example below demonstrates the suggested approach. Uncomment the second styles portion to see the difference.

````CSHTML

> @*Center the LoaderContainer in the viewport*@

Reviewer: ```suggestion
@* Center the LoaderContainer in the viewport *@
```

<style>

>     .customized-loader-container.k-loader-container{

Reviewer: ```suggestion
    .customized-loader-container.k-loader-container {
```

        position: fixed;
    }

> /*    .customized-loader-container .k-loader-container-inner{

Reviewer: ```suggestion
    /* .customized-loader-container .k-loader-container-inner {
```
